geoh5py/data/color_map.py

```diff
@@ -50,27 +61,39 @@ def values(self) -> np.ndarray:
         where V (Values) are sorted floats defining the position of each RGBA.
         R (Red), G (Green), B (Blue) and A (Alpha) are integer values between [0, 255].
         """
-        return self._values
+        if self._values is None:
+            return self._values
+        return np.vstack([self._values[name] for name in self._names])
 
     @values.setter
     def values(self, values: np.ndarray):
-        names = ["Value", "Red", "Green", "Blue", "Alpha"]
-        formats = ["<f8", "u1", "u1", "u1", "u1"]
 
-        if isinstance(values.dtype, np.dtype):
-            assert values.dtype.names is not None and all(
-                name in names for name in (values.dtype.names)
-            ), f"Input 'values' must contain fields with types {names}"
-            self._values = np.asarray(values, dtype=list(zip(names, formats)))
+        if not isinstance(values, np.ndarray):
+            raise TypeError(f"Input 'values' of ColorMap must be of type {np.ndarray}.")
+
+        if np.issubdtype(values.dtype.base, np.number):
+            if values.shape[1] != 5:
+                raise ShapeValidationError("values", values.shape, "(*, 5)")
```

Comment:
This is great for now, but we should think about how we would incorporate the larger validation framework in here.  I think this might point to having a lightweight constructor to support validations per param.  That way we could construct a validation here with a dictionary containing the types and shape rules and then call the .validate method to accomplish these two if condition: validate patterns.

Reply:
I'll add that the use case here should be incorporated into the shape validator, ie: pass in one or more free dimensions and enforce the rest.